core/src/backends/lnd/rest/types.rs

```diff
@@ -253,3 +255,65 @@ impl TryInto<PayInvoiceResult> for SendPaymentSyncResponse {
         Ok(result)
     }
 }
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct DecodeInvoiceResponse {
+    pub destination: String,
+    pub payment_hash: String,
+    pub num_satoshis: String,
+    pub timestamp: String,
+    pub expiry: String,
+    pub description: String,
+    pub description_hash: String,
+    pub fallback_addr: String,
+    pub cltv_expiry: String,
+    pub route_hints: Vec<Value>,
+    pub payment_addr: String,
+    pub num_msat: String,
+    pub features: HashMap<u16, DecodeInvoiceFeature>,
+}
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
```

Comment:
```suggestion
#[derive(Default, Debug, Clone, PartialEq, Deserialize)]
```
We don't need the `Serialize` trait here as we only get that type in a response to a request.